moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -896,29 +896,49 @@ bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext &cont
     ROS_WARN_NAMED("traj_execution", "Failed to validate trajectory: couldn't receive full current joint state within 1s");
     return false;
   }
+  current_state->enforceBounds();
 
+  // build a RobotState that can be used to compare current_state and the start state with enforced bounds
+  robot_state::RobotState bounded_start_state(*current_state);
   for (std::vector<moveit_msgs::RobotTrajectory>::const_iterator traj_it = context.trajectory_parts_.begin();
        traj_it != context.trajectory_parts_.end(); ++traj_it)
   {
     const std::vector<double> &positions = traj_it->joint_trajectory.points.front().positions;
     const std::vector<std::string> &joint_names = traj_it->joint_trajectory.joint_names;
     const std::size_t n = joint_names.size();
+
     if (positions.size() != n)
     {
       ROS_ERROR_NAMED("traj_execution", "Wrong trajectory: #joints: %zu != #positions: %zu", n, positions.size());
       return false;
     }
 
+    try
+    {
+      bounded_start_state.setVariablePositions(joint_names, positions);
```

Comment:
This will not work as soon as the trajectory is split into parts because several controllers will be used for different subsets of joints. I will suggest an alternative.